.github/workflows/build-node-packages.yml

```diff
@@ -0,0 +1,202 @@
+      shell: pwsh
+
+    - name: Publish artifact
+      uses: actions/upload-artifact@v2
+      with:
+        name: node-${{ env.VERSION }}-${{ matrix.platform }}
+        path: /home/runner/work/node-versions/artifact
+
+  test_node:
+    name: Test Node ${{ github.event.inputs.VERSION }} ${{ matrix.platform }}
+    needs: build_node
+    runs-on: ${{ matrix.os }}-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        - os: ubuntu
+          platform: linux
+        - os: macos
+          platform: darwin
+        - os: windows
+          platform: win32
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: true
+
+    - name: Set AGENT_TOOLSDIRECTORY variable
+      if: matrix.platform == 'win32'
+      run: |
+        # GitHub Windows images don't have `AGENT_TOOLSDIRECTORY` variable
+        echo ::set-env name=AGENT_TOOLSDIRECTORY::$RUNNER_TOOL_CACHE
+      shell: bash
+
+    - name: Fully cleanup the toolcache directory before testing
+      run: |
+        ./helpers/clean-toolcache.ps1 -ToolName "node"
+      shell: pwsh
+
+    - name: Download artifact
+      uses: actions/download-artifact@v2
+      with:
+        path: ${{ runner.temp }}
+
+    - name: Extract files
+      run: |
+        if ('${{ matrix.platform }}' -eq 'win32') {
+          $artifactName = "node-${{ env.VERSION }}-${{ matrix.platform }}-${{ env.ARCHITECTURE }}.7z"
+          7z.exe x "$artifactName" -y | Out-Null 
+        } else {
+          $artifactName = "node-${{ env.VERSION }}-${{ matrix.platform }}-${{ env.ARCHITECTURE }}.tar.gz"
+          tar -xzf $artifactName
+        }
+      working-directory: ${{ runner.temp }}/node-${{ env.VERSION }}-${{ matrix.platform }}
+      shell: pwsh
+
+    - name: Apply build artifact to the local machine
+      run: |
+        if ('${{ matrix.platform }}' -eq 'win32') { powershell ./setup.ps1 } else { sh ./setup.sh }
+      working-directory: ${{ runner.temp }}/node-${{ env.VERSION }}-${{ matrix.platform }}
+      shell: pwsh
+
+    - name: Setup node ${{ env.VERSION }}
+      uses: actions/setup-node@v2.1.1
+      with:
+        node-version: ${{ env.VERSION }}
+
+    - name: Wait for the logs
+      run: |
+        Write-Host "Fake step that do nothing"
+        Write-Host "We need it because log of previous step 'Setup Node' is not available here yet."
+        Write-Host "In testing step (Node.Tests.ps1) we analyze build log of 'Setup Node' task"
+        Write-Host "to determine if Node.js version was consumed from cache and was downloaded"
+      shell: pwsh
+
+    - name: Run tests
+      run: |
+        Install-Module Pester -Force -Scope CurrentUser -RequiredVersion 4.10.1
+        Import-Module Pester
+        $pesterParams = @{
+          Path="./Node.Tests.ps1";
+          Parameters=@{
+            Version="$env:VERSION";
+          }
+        }
+        Invoke-Pester -Script $pesterParams -EnableExit -OutputFile "test_results.xml" -OutputFormat NUnitXml
```

Comment:
Why we pass `OutputFile` and `OutputFormat` if we don't use it.
Is it possible to switch to Pester 5 Btw?

Reply:
I've removed options.
Need to update to Pester 5

Reply:
Updated